### PR TITLE
python3Packages.readmdict: init at 0.1.1

### PR DESCRIPTION
**Pull request by paveloom**
## Description of changes

https://github.com/ffreemt/readmdict

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/python-modules/readmdict/default.nix [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+  src = fetchPypi {
+    inherit pname version;
+    hash = "sha256-RVVvXOR2tnPN+ab5Ak7YHLbgGq7AE5Es7cFJqjvTvrY=";
+  };
+
+  postPatch = ''
+    substituteInPlace pyproject.toml \
+      --replace "poetry>=0.12" "poetry-core" \
+      --replace "poetry.masonry" "poetry.core.masonry"
+  '';
```
Comment (pbsds) [inline pkgs/development/python-modules/readmdict/default.nix:26]: Please submit an issue/pull request about this fix upstream and link to it

File: pkgs/development/python-modules/readmdict/default.nix [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+
+  nativeBuildInputs = [
+    poetry-core
+  ];
+
+  propagatedBuildInputs = [
+    python-lzo
+  ];
+
```
Comment (pbsds) [inline pkgs/development/python-modules/readmdict/default.nix:35]:
```suggestion

  nativeCheckInputs = [
    pytestCheckHook
  ]

```

**Review by pbsds — CHANGES_REQUESTED** [review]

File: pkgs/development/python-modules/readmdict/default.nix (hunk 2 above)
Comment (paveloom) [reply pkgs/development/python-modules/readmdict/default.nix:36]: The runner can't find any:

```
python3.10-readmdict> Executing pytestCheckPhase
python3.10-readmdict> ============================= test session starts ==============================
python3.10-readmdict> platform linux -- Python 3.10.12, pytest-7.4.0, pluggy-1.2.0
python3.10-readmdict> rootdir: /build/readmdict-0.1.1
python3.10-readmdict> collected 0 items
python3.10-readmdict> ============================ no tests ran in 0.00s =============================
python3.10-readmdict> /nix/store/vr6wwdxkmyy44sg0gwxi10b8fc5zhwz0-stdenv-linux/setup: line 1596: pop_var_context: head of shell_variables not a function context
error: builder for '/nix/store/vascw48f6bm1r0nm8dvjyxi9wkjzh7h6-python3.10-readmdict-0.1.1.drv' failed with exit code 5;
       last 10 log lines:
       > Check whether the following modules can be imported: readmdict
       > pytestCheckPhase
       > Executing pytestCheckPhase
       > ============================= test session starts ==============================
       > platform linux -- Python 3.10.12, pytest-7.4.0, pluggy-1.2.0
       > rootdir: /build/readmdict-0.1.1
       > collected 0 items
       >
       > ============================ no tests ran in 0.00s =============================
       > /nix/store/vr6wwdxkmyy44sg0gwxi10b8fc5zhwz0-stdenv-linux/setup: line 1596: pop_var_context: head of shell_variables not a function context
       For full logs, run 'nix log /nix/store/vascw48f6bm1r0nm8dvjyxi9wkjzh7h6-python3.10-readmdict-0.1.1.drv'.
```

[There is one](https://github.com/ffreemt/readmdict/blob/967d91427174a38104d03354bec58345cf78d7f4/tests/test_version.py), however, but it just check the version.

Comment (paveloom) [reply pkgs/development/python-modules/readmdict/default.nix:36]: Checked the archive from PyPI, there aren't any tests packaged.

Comment (pbsds) [reply pkgs/development/python-modules/readmdict/default.nix:36]: The test might be trivial, but upstream may chose to add more in the future. Consider using `fetchFromGitHub`

**Comment by pbsds** [comment]
In a `--pure` nix-shell i get this error:
```
$ readmdict
Traceback (most recent call last):
  File "/nix/store/545p7qvfy959mbs9962jg8c6bkrcs80j-python3.10-readmdict-0.1.1/bin/.readmdict-wrapped", line 9, in <module>
    sys.exit(main())
  File "/nix/store/545p7qvfy959mbs9962jg8c6bkrcs80j-python3.10-readmdict-0.1.1/lib/python3.10/site-packages/readmdict/__main__.py", line 47, in main
    import tkinter
  File "/nix/store/pzf6dnxg8gf04xazzjdwarm7s03cbrgz-python3-3.10.12/lib/python3.10/tkinter/__init__.py", line 37, in <module>
    import _tkinter # If this fails your Python may not be configured for Tk
ModuleNotFoundError: No module named '_tkinter'
```

**Comment by paveloom** [comment]
Oh, didn't notice they had a CLI binary.

**Comment by pbsds** [comment]
The parentheses in the `meta.description` is questionable, but i'd say it LGTM.

---

Result of `nixpkgs-review pr 257437` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>4 packages built:</summary>
  <ul>
    <li>readmdict (python310Packages.readmdict)</li>
    <li>readmdict.dist (python310Packages.readmdict.dist)</li>
    <li>python311Packages.readmdict</li>
    <li>python311Packages.readmdict.dist</li>
  </ul>
</details>

---

- [x] package path fits guidelines
- [x] package name fits guidelines
- [x] package version fits guidelines
- [x] package build on x86_64-linux
- [x] executables tested on x86_64-linux
- [?] `meta.description` is set and fits guidelines
- [x] `meta.license` fits upstream license
- [x] `meta.maintainers` is set
- [x] build time only dependencies are declared in `nativeBuildInputs`
- [x] source is fetched using the appropriate function
- [x] the list of `phases` is not overridden
- [x] when a phase (like `installPhase`) is overridden it starts with `runHook preInstall` and ends with `runHook postInstall`.
- [x] patches have a comment describing either the upstream URL or a reason why the patch wasn't upstreamed
- [x] patches that are remotely available are fetched rather than vendored

**Review by pbsds — APPROVED** [review]

**Pull request merged**